### DISCUSSION TITLE
[testability] Add the ability to reassign user relationships to current or specified user

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5640] [Feature Requests]
<img width="684" alt="image" src="https://user-images.githubusercontent.com/37256050/189446201-77ef511a-ec6e-435b-96a8-85cb0c3b7225.png">

Allowing users to delete specific users by linking to the blocking records is a great idea! It comes with its own risks, especially in cases where an old admin or collection manager retires.

You can see that very important resources are often associated with them (trees, schema mappings, etc):

<img width="540" alt="image" src="https://user-images.githubusercontent.com/37256050/189446323-d2326310-ecb3-4fe0-a72c-fd75bb7240b5.png">

It would be nice to replace references to that user with your current or specified user account.

**Comment by maxpatiiuk** [comment]
This issue is specific to deleting SpecifyUser resources only.

But there is a solution that is not specific to SpecifyUser - open the associated record and assign a different user.

That field does not appear to be present on the screenshot above, which would be fixed by this - https://github.com/specify/specify7/issues/2105


Also, I don't think it always makes sense to reassign the resources to current user, as for some resources it only makes sense for a user to have a single resource (UserPreferences)
